### DISCUSSION TITLE
[Theory or Implementation] Limitations of Linear Wave Theory

### DISCUSSION THREAD

**Discussion by pm-gusmano** [WEC-Sim/WEC-Sim#1399] [Q&A] [answered]
**Is your question request related to a problem? Please describe.**
N/A

**Describe the theory or implementation approach that you have a question**
I'd be very thankful for an explanation on WEC-Sim's limits, especially as it relates to modeling WEC responses to extreme sea states.

**Personal project context**
My team and I are interested in modifying the example RM3 to increase its survivability and PTO Power absorption. It seems the easiest way to do this is by developing power matrices (and some sort of equivalent for a metric representing survivability) for each variation in our design, and use the joint probability distribution matrix found in _Neary et. al (2014). Methodology for Design and Economic Analysis of Marine Energy Conversion (MEC) Technologies._ to compare average power and survivability. To trust this methodology, I need to know that WEC-Sim can accurately handle each simulation, especially those in extreme sea states. Figure 2 of _Coe, R. G., & Neary, V. S. (2014). Review of Methods for Modeling Wave Energy Converter Survival in Extreme Sea States._ (not posted here because I don't know how legal that is) seems to suggest that, with g = 9.81 m/s^2 and infinitely deep water, we'll get the following profile for the limits of linear wave theory.

![image](https://github.com/user-attachments/assets/e94ec2da-2d27-483b-b6e9-d0a291b3fc1c)

At the same time, I've seen examples in `WEC-Sim_Applications` that seem to go beyond this curve, which is why I've submitted some clarification ‒ what are the limits of linear wave theory?

**Describe the type of conclusion or resolution you are looking for**
I need to know whether I can use the methodology outlined above to assess the power absorption and survivability of my modified RM3s, which depends on whether I can trust WEC-Sim's responses in extreme sea states.

**Additional context**
N/A

**Comment by MShabara** [comment, marked as answer]
Hi @Gusmano-2-OSU 

It's great to see that a research group in The OSU is using WEC-Sim, and thank you for your question and for highlighting this issue.

Some examples in `WEC-Sim_Applications` may simulate conditions beyond the strict linear regime. However, these simulations are typically intended for conceptual or sensitivity analyses rather than detailed validation of extreme sea states.

---

## WEC-Sim's Capabilities

WEC-Sim is a low-to-mid fidelity solver and supports some non-linear effects. 

### Key non-linear capabilities include:
- **[Non-linear hydrodynamic forces](https://wec-sim.github.io/WEC-Sim/dev/user/advanced_features.html#nonlinear-buoyancy-and-froude-krylov-excitation)**: Enabled via the `body(i).nonlinearHydro=1` or `2` option. Note: Use finely meshed `.stl` file if this option is enabled.
- **Second order excitation forces**: Supported through QTF coefficients from solvers like WAMIT or NEMOH.
- **Variable hydro**: A new feature that allows users to simulate large motions or other non-linear effects by dynamically updating hydrodynamic coefficients during the simulation when certain conditions are met.

However, WEC-Sim cannot directly model breaking waves, highly non-linear forces.

---

## Recommendations

If you're unsure whether your model is linear or non-linear, test `body(i).nonlinearHydro=2` and `body(i).nonlinearHydro=0`. If the results change drastically, non-linear effects cannot be ignored.

---

## Additional Thoughts

It may be worth developing a feature in WEC-Sim to flag when simulations exceed linear theory assumptions based on your provided reference. This could improve user awareness.

Let me know if you’d like further clarification or have additional questions!
